components/eam/src/chemistry/mozart/lin_strat_chem.F90

```diff
@@ -595,12 +596,20 @@ subroutine linv3_strat_chem_solve(ncol, lchnk, xvmr, h2ovmr, xsfc, o3col,temp, s
              end if
           !
           ! update vmr
-           xvmr(i,k,  o3lnz_ndx) =   o3_new
-           xvmr(i,k, n2olnz_ndx)   = n2o_new
-           xvmr(i,k, noylnz_ndx)   = noy_new
-           xvmr(i,k, ch4lnz_ndx)   = ch4_new
-           xvmr(i,k, h2olnz_ndx)   = h2o_new
-          
+           !kzm ++
+           !xvmr(i,k,  o3lnz_ndx) =   o3_new
+           !xvmr(i,k, n2olnz_ndx)   = n2o_new
+           !xvmr(i,k, noylnz_ndx)   = noy_new
+           !xvmr(i,k, ch4lnz_ndx)   = ch4_new
+           !xvmr(i,k, h2olnz_ndx)   = h2o_new
```

Comment:
The above commented out block can be removed altogether, probably replacing with a descriptive comment why applying if-conditional check: as a defense the assignments are only performed when the species are active. Otherwise the index would be an invalid value (-1).

Reply:
@wlin7 changed as suggested.